### DISCUSSION TITLE
Understanding CouchDB logs

### DISCUSSION THREAD

**Discussion by gboer** [apache/couchdb#3902] [Q&A] [answered]
I'm trying to understand the CouchDB logs and what each field means. I've searched a number of times on this, but as far as I know, it doesn't exist in the documentation, nor have I found it anywhere else online.

An example line:
`[notice] 2022-01-17T11:55:21.745418Z nonode@nohost <0.16928.147> b83546de67 staging-couchdb:5984 172.19.0.2 JsQHfyB3UbV2LBIpomV6ek5DIhwu7OQd POST /db-test-17e67e41e56 500 ok 40`

These are the fields I understand so far:

1. log level
2. timestamp
3. user@host, but I don't really understand what this is. As far as I can see, it's always the same.
4. No idea?
5. Request ID / connection ID
6. hostname:port
7. IP of client
8. Session ID?
9. HTTP method
10. HTTP path
11. HTTP error code
12. ok .. seems to always be ok?
13. Response times in milliseconds.

Would be great if someone can confirm and/or add the missing bits (number 3, 4, 8, and 12).

**Comment by nickva** [comment, marked as answer]
Good question @gboer I don't know if we had documented it anywhere.

Most of your guesses are correct.

The first 5 parameters are generated in [couch_log](https://github.com/apache/couchdb/blob/4e9c5588d765b84742784de5aafa146d14eed11f/src/couch_log/src/couch_log_writer_stderr.erl#L38-L48). The rest are from the http request handling section in [chttpd](https://github.com/apache/couchdb/blob/4e9c5588d765b84742784de5aafa146d14eed11f/src/chttpd/src/chttpd.erl#L493-L502) :  
   1. Log level
   2. Timestamp
   3. Erlang node name where this entry was generated. For single node without erlang distribution enabled it would be `nonode@nohost`. If you have a cluster setup it might be `node1@internal.couch.net` for example
   4. Erlang process ID which generated the log entry.
   5. A randomly generated request ID / nonce which can be used to track requests or failures across a load balancer. It gets returned as the `"X-Couch-Request-ID"` response header.
   6. Request header `Host:` value
   7. Client IP (Peer IP)
   8. Authenticated user / session ID. This is url encoded. Might also be `undefined` admin requests.
   9. HTTP method
   10. HTTP path
   11. HTTP return code
   12. Request status `ok` would be normal, might have other values perhaps `aborted` if request was suddenly aborted in the middle of processing
   13. Request time in milliseconds

**Reply by gboer** [reply]
Thanks for your reply @nickva!

I couldn't find it anywhere in the documentation and I've searched multiples times. So I'm glad I finally know what's what now :)
